app/controllers/api/v2/config_reports_controller.rb

```diff
@@ -40,7 +40,10 @@ def show
 
       def create
         @config_report = ConfigReport.import(params.to_unsafe_h[:config_report], detected_proxy.try(:id))
-        process_response @config_report.errors.empty?
+        errors = @config_report.errors.empty?
+        # RABL is extremely slow for huge reports and ENC script does not care about the output
```

Comment:
This is confusing. The ENC script has nothing to do with config reports.

Also note that the config reports controller is not just for Puppet reports. It's also used by Ansible and possibly others. It's breaking the API and as such it must at least be accompanied by an upgrade warning.

I think this may deserve its own issue so it properly shows up in the changelog.

Reply:
I state above that the change is breaking the API and I know that other plugins use this endpoint. True this should be split up from this PR. Luckily it is completely isolated and does not depend on hash-index change. Pulling it into a separate PR (note for me sha `62cbab1`)

Reply:
Removing from there, the PR is: https://github.com/theforeman/foreman/pull/7987